src/cdn_definitions/schema.json

```diff
@@ -7,6 +7,124 @@
             "pattern": "^/",
             "type": "string"
         },
+        "arch": {
+            "type": "string"
+        },
+        "arch_list": {
+            "items": {
+                "$ref": "#/definitions/arch"
+            },
+            "type": "array",
+            "uniqueItems": true
+        },
+        "beta_and_ga_keys_mapping": {
+            "additionalProperties": false,
+            "properties": {
+                "beta_keys": {
+                    "$ref": "#/definitions/signing_key_list"
+                },
+                "ga_keys": {
+                    "$ref": "#/definitions/signing_key_list"
+                }
+            },
+            "required": [
+                "ga_keys"
+            ],
+            "type": "object"
+        },
+        "cfme_version_mappings_list": {
+            "additionalProperties": false,
+            "patternProperties": {
+                ".*": {
+                    "$ref": "#/definitions/major_or_release_version"
+                }
+            },
+            "propertyNames": {
+                "$ref": "#/definitions/major_or_release_version"
+            },
+            "type": "object"
+        },
+        "current_latest_rhel_versions": {
+            "additionalProperties": false,
+            "patternProperties": {
+                ".*": {
+                    "anyOf": [
+                        {
+                            "type": "null"
+                        },
+                        {
+                            "$ref": "#/definitions/release_version"
+                        }
+                    ]
+                }
+            },
+            "propertyNames": {
+                "pattern": "^(current|latest)\\_rhel[0-9]+\\_(beta|dist)$",
+                "type": "string"
+            },
+            "type": "object"
+        },
+        "env_to_releasever_mapping": {
+            "additionalProperties": false,
```

Comment:
Hmm, I don't think this is correct because we do actually make it possible right now for the user of cdn-utils to define their own environments, so "qa", "stage", "prod" aren't the only values.  Search code for /etc/rcm/cdn-environments.json.  I think the ability should be retained - it's preferable for testers to e.g. define their own test1, test2 environment rather than having to *overwrite* one of the existing environments if they want to test against their own infrastructure.

Can we make this one a patternProperties ".*" to allow arbitrary environments?  "required" qa, stage, prod may still make sense though.

Reply:
+1, populating the prod, stage, and qa mappings would be additional work for testers, so I'll also remove the envs from the required properties list. This sort of requirement could instead be enforced via a test in cdn-defintions-private. I'll leave the envs in data.yaml as an example.